### ISSUE TITLE
Way to get the next time the cron would run

### ISSUE THREAD

**Issue by maxharlow** [bradymholt/cRonstrue#41]
A `getNext` method that returned a human-readable date when it would run next would be really useful. Would also mean feature parity with [PrettyCron](https://github.com/azza-bazoo/prettycron).

**Comment by bradymholt** [comment]
Thanks for the suggestion but adding this type of functionality would be non-trivial (I've estimated it could double the size of the code base) and fall outside the scope of the original intention of the library.  There are some npm packages out there that do this just fine such as https://www.npmjs.com/package/cron-parser.  If you need this functionality, I'd recommend using one of those packages to get the "next" interval and cRonstrue to describe the expression.  If you really want a single library interface for both you could create another package that wraps both of these libraries and has a single/common interface.

**Comment by nick** [comment]
If anyone else comes across this... I'd recommend the npm package [later](https://www.npmjs.com/package/later)

**Comment by Trial-In-Error** [comment]
Just chiming in for other people that arrive here:
As far as I can tell, for some particular use cases, there aren't currently good options. Note that, for the list below, if I found a reason to disqualify it from my search, I didn't include all the things it might be lacking (e.g., a "doesn't support years" entry may also not support W or L, and I did not confirm):

- `cron-parser` doesn't support the W or L syntaxes, and is incapable of handling nextEvents that contain a year.
- `later` is 5 years unsupported, and doesn't play well with webpack.
- `node-cron` is intended only for scheduling on the server-side, and doesn't play well with any non-node environment (despite technically having deferred jobs that provide a "nextDates" array).
- `prettyCron` is LGPL licensed.
- `cron-converter` doesn't support years.
- `croner` doesn't support years.
- `cron-time` doesn't support years.
- `node-schedule` doesn't support years.

In short, one of the nice things about cRonstrue is that it isn't build ontop of 'cron-parser', which many other libraries are, and it's got W/L/year support as part of this.
